kobo/apps/audit_log/models.py

```diff
@@ -204,4 +211,63 @@ def create_from_request(
         # add any other metadata the caller may want
         if extra_metadata is not None:
             metadata.update(extra_metadata)
+        if is_submission:
+            # return a SubmissionAccessLog specifically to trigger the post_save hook
+            return SubmissionAccessLog.objects.create(
+                user=logged_in_user, metadata=metadata
+            )
         return AccessLog.objects.create(user=logged_in_user, metadata=metadata)
+
+
+class SubmissionGroupManager(AccessLogManager):
+    def get_queryset(self):
+        return (
+            super()
+            .get_queryset()
+            .filter(metadata__auth_type=ACCESS_LOG_SUBMISSION_GROUP_AUTH_TYPE)
+        )
+
+    def create(self, **kwargs):
+        metadata = kwargs.pop('metadata', {})
+        metadata['auth_type'] = ACCESS_LOG_SUBMISSION_GROUP_AUTH_TYPE
+        kwargs['metadata'] = metadata
+        return super().create(**kwargs)
+
+
+class SubmissionGroup(AccessLog):
+    objects = SubmissionGroupManager()
+
+    class Meta:
+        proxy = True
+
+
+class SubmissionAccessLogManager(AccessLogManager):
+    def create(self, **kwargs):
+        metadata = kwargs.pop('metadata', {})
+        metadata['auth_type'] = ACCESS_LOG_SUBMISSION_AUTH_TYPE
+        kwargs['metadata'] = metadata
+        return super().create(**kwargs)
+
+
+class SubmissionAccessLog(AccessLog):
+    objects = SubmissionAccessLogManager()
+
+    class Meta:
+        proxy = True
+
+    def add_to_existing_submission_group(
+        self, submission_group: SubmissionGroup
+    ):
+        if self.user_uid != submission_group.user_uid:
+            logging.error(
+                f'Cannot add submission log with user {self.user_uid} to group with user {submission_group.user_uid}'
+            )
+            return
+        self.submission_group = submission_group
```

Comment:
I go back and forth on who should be responsible for saving the model. There's a reasonable chance any future code that calls this method would make multiple changes and it would be a waste of db work to save every time. I don't feel strongly about it though.

Reply:
On further reflection, after a refactor, ended up deleting this method all together